feat: add support for accessor arrays and refactor `stats/base/mean`

gururaj1512 opened the pull request:
Resolves #5629.

## Description

> What is the purpose of this pull request?

This pull request:

-   update `stats/base/mean` to follow latest project conventions
-   add support for accessor arrays

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   No

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add support for accessor arrays and refactor `stats/base/mean`

kgryte commented:
@gururaj1512 Was there a reason to create a whole new PR when https://github.com/stdlib-js/stdlib/pull/5730 already exists and looks to only require minor modification?

gururaj1512 committed:
fix: examples and benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 commented:
> @gururaj1512 Was there a reason to create a whole new PR when #5730 already exists and looks to only require minor modification?

Actually the PR had diffrent implementation and no added testcases. I had mistakenly worked upon same thought it would be okay to create it. Should I close this?
Appologies due to negligence

kgryte commented:
@gururaj1512 Yeah, that PR incorrectly refactored away from using `meanpn`. That change should be reverted in that PR, but I feel fairly confident that you should be able to clean up that PR rather than creating a new one like you are doing here.

gururaj1512 commented:
Got it. Actually it would be unfair to close someones PR which is worked on even if it is partial. I'll take care of it further.
